website/public/js/controllers/guildsCtrl.js

```diff
@@ -4,91 +4,110 @@ habitrpg.controller("GuildsCtrl", ['$scope', 'Groups', 'User', 'Challenges', '$r
+            .then(function (response) {
+              var createdGroup = response.data.data;
+              if (createdGroup.privacy == 'public') {
+                Analytics.track({'hitType':'event', 'eventCategory':'behavior', 'eventAction':'join group', 'owner':true, 'groupType':'guild', 'privacy': createdGroup.privacy, 'groupName':createdGroup.name})
+              } else {
+                Analytics.track({'hitType':'event', 'eventCategory':'behavior', 'eventAction':'join group', 'owner':true, 'groupType':'guild', 'privacy': createdGroup.privacy})
+              }
+              $rootScope.hardRedirect('/#/options/groups/guilds/' + createdGroup._id);
+            });
         }
       }
 
-      $scope.join = function(group){
-        // If we're accepting an invitation, we don't have the actual group object, but a faux group object (for performance
-        // purposes) {id, name}. Let's trick ngResource into thinking we have a group, so we can call the same $join
-        // function (server calls .attachGroup(), which finds group by _id and handles this properly)
+      $scope.join = function (group) {
+        var groupId = group._id;
+
+        //  If we don't have the _id property, we are joining from an invitation
+        //  which contains a id property of the group
         if (group.id && !group._id) {
-          group = new Groups.Group({_id:group.id});
+          groupId = group.id;
         }
 
-        group.$join(function(joined){
-          if (joined.privacy == 'public') {Analytics.track({'hitType':'event','eventCategory':'behavior','eventAction':'join group','owner':false,'groupType':'guild','privacy':joined.privacy,'groupName':joined.name})}
-          else {Analytics.track({'hitType':'event','eventCategory':'behavior','eventAction':'join group','owner':false,'groupType':'guild','privacy':joined.privacy})}
-          $rootScope.hardRedirect('/#/options/groups/guilds/' + joined._id);
-        })
+        Groups.Group.join(groupId)
+          .then(function (response) {
+            var joinedGroup = response.data.data;
+
+            if (joinedGroup.privacy == 'public') {
+              Analytics.track({'hitType':'event', 'eventCategory':'behavior', 'eventAction':'join group', 'owner':false, 'groupType':'guild','privacy': joinedGroup.privacy, 'groupName': joinedGroup.name})
+            } else {
+              Analytics.track({'hitType':'event', 'eventCategory':'behavior', 'eventAction':'join group', 'owner':false, 'groupType':'guild','privacy': joinedGroup.privacy})
+            }
+            $rootScope.hardRedirect('/#/options/groups/guilds/' + joinedGroup._id);
+          });
+      }
+
+      $scope.reject = function(guild) {
+        Groups.Group.rejectInvite(guild.id);
+        var i = _.findIndex(User.user.invitations.guilds, {id: guild.id});
+        if (i !== -1) {
+          User.user.invitations.guilds.splice(i, 1);
+          User.set({'invitations.guilds':User.user.invitations.guilds});
```

Comment:
which seems to be used at line 60 and should take care of removing the invite from the user document